ssv/committee_runner.go

```diff
@@ -191,14 +204,22 @@ func (cr CommitteeRunner) ProcessPostConsensus(signedMsg *types.PartialSignature
 		}
 
 		for _, validator := range validators {
+			// Skip if no quorum
+			if !cr.BaseRunner.State.PostConsensusContainer.HasQuorum(validator, root) {
+				continue
+			}
+			// Skip if already submitted
+			if cr.HasSubmitted(role, validator, slot) {
+				continue
+			}
```

Comment:
I think that if we have already submitted for the root and validator `basePostConsensusMsg` shouldn't add it again... but maybe it is a check worth having?

This is because:

```
	if hasQuorum && !prevQuorum {
			// Notify about first quorum only
			roots = append(roots, msg.SigningRoot)
			anyQuorum = true
		}
```

Reply:
Yes.
`basePostConsensusMsgProcessing` returns a list of roots that got a quorum. But then, `findValidators` returns which validators are associated with a certain root.
But, notice that a validator X can submit for root R while validator Y haven't submitted yet. So, R may be returned again by `basePostConsensusMsgProcessing` and `findValidators` will return validator X. The check handles such cases.